src/components/vm/vmActions.jsx

```diff
@@ -53,9 +56,159 @@ import {
+    store.dispatch(
+        updateVm({
+            connectionName: vm.connectionName,
+            name: vm.name,
+            error: {
+                text: cockpit.format(_("VM $0 failed to pause"), vm.name),
+                detail: ex.message,
+            }
+        })
+    );
+});
+const onResume = (vm) => domainResume({ name: vm.name, id: vm.id, connectionName: vm.connectionName }).catch(ex => {
+    store.dispatch(
+        updateVm({
+            connectionName: vm.connectionName,
+            name: vm.name,
+            error: {
+                text: cockpit.format(_("VM $0 failed to resume"), vm.name),
+                detail: ex.message,
+            }
+        })
+    );
+});
+const onForceoff = (vm) => domainForceOff({ name: vm.name, id: vm.id, connectionName: vm.connectionName })
+        .then((vm) => !vm.persistent && cockpit.location.go(["vms"]))
+        .catch(ex => {
+            store.dispatch(
+                updateVm({
+                    connectionName: vm.connectionName,
+                    name: vm.name,
+                    error: {
+                        text: cockpit.format(_("VM $0 failed to force shutdown"), vm.name),
+                        detail: ex.message,
+                    }
+                })
+            );
+        });
+const onSendNMI = (vm) => domainSendNMI({ name: vm.name, id: vm.id, connectionName: vm.connectionName }).catch(ex => {
+    store.dispatch(
+        updateVm({
+            connectionName: vm.connectionName,
+            name: vm.name,
+            error: {
+                text: cockpit.format(_("VM $0 failed to send NMI"), vm.name),
+                detail: ex.message,
+            }
+        })
+    );
+});
+
+export const ConfirmModal = ({ idPrefix, actionsList, startTime, title, titleIcon }) => {
+    const Dialogs = useDialogs();
+
+    const actions = actionsList.map(action =>
+        <Button variant={action.variant}
+            key={action.id}
+            id={`${idPrefix}-${action.id}`}
+            onClick={() => {
+                action.handler();
+                Dialogs.close();
+            }}>
+            {action.name}
+        </Button>
+    );
+    actions.push(
+        <Button variant="link" key="cancel" onClick={Dialogs.close}>
+            {_("Cancel")}
+        </Button>
+    );
+
+    return (
+        <Modal id={`${idPrefix}-confirm-action-modal`}
+            position="top"
+            variant="small"
+            onClose={Dialogs.close}
+            title={title}
+            titleIconVariant={titleIcon}
+            isOpen
+            footer={actions}>
+            <DescriptionList isHorizontal>
+                <DescriptionListGroup id="uptim">
+                    <DescriptionListTerm>{_("Uptime")}</DescriptionListTerm>
+                    <DescriptionListDescription>{distanceToNow(new Date(startTime))}</DescriptionListDescription>
```

Comment:
This is not correct for remote connections. The startTime is from the server running the VM. distanceToNow will run new Date() on the browser, which can be on a remote server, different than the one running the VM.

Reply:
You need to run the date command on the VM's server.

Reply:
Oh right. 
But turns out, it's even easier than that. Libvirt always uses +0000 (GMT+0) timezone for times in the logs. Therefore even when used in the browser, `new Date("2023-05-05 11:22:03.043+0000")` will figure out the correct time zone based off the "+0000" in the end of the string and adjust it to the browser's time zone.
So I just updated the the API where I parse the time from the logs, to pass the whole timestamp even with timezone, and everything else should work automagically.